apps/namadillo/src/App/Ibc/IbcFromNamada.tsx

```diff
@@ -0,0 +1,162 @@
+import { Key as KeplrKey, Window as KeplrWindow } from "@keplr-wallet/types";
+import { Panel } from "@namada/components";
+import { WindowWithNamada } from "@namada/types";
+import BigNumber from "bignumber.js";
+import { getSdkInstance } from "hooks";
+import { useAtomValue } from "jotai";
+import { useEffect, useState } from "react";
+
+import { NamCurrency } from "App/Common/NamCurrency";
+import { NamInput } from "App/Common/NamInput";
+import { accountBalanceAtom, defaultAccountAtom } from "atoms/accounts";
+import { chainAtom, chainParametersAtom } from "atoms/chain";
+import { getFirstError } from "atoms/utils";
+
+const keplr = (window as KeplrWindow).keplr!;
+const namada = (window as WindowWithNamada).namada!;
+
+const keplrChain = "theta-testnet-001";
+
+const buttonStyles = "bg-white my-2 p-2 block";
+
+export const IbcFromNamada: React.FC = () => {
+  const [error, setError] = useState("");
+  const [keplrAccount, setKeplrAccount] = useState<KeplrKey>();
+  const [amount, setAmount] = useState<BigNumber>();
+  const [channel, setChannel] = useState("");
+
+  const namadaAccount = useAtomValue(defaultAccountAtom);
+  const balance = useAtomValue(accountBalanceAtom);
+  const namadaChain = useAtomValue(chainAtom);
+  const namadaChainParams = useAtomValue(chainParametersAtom);
+
+  useEffect(() => {
+    const error = getFirstError(
+      namadaAccount,
+      balance,
+      namadaChain,
+      namadaChainParams
+    );
+    setError(error ? error.message : "");
+  }, [
+    namadaAccount.isError,
+    balance.isError,
+    namadaChain.isError,
+    namadaChainParams.isError,
+  ]);
+
+  const withErrorReporting =
+    (fn: () => Promise<void>): (() => Promise<void>) =>
+    async () => {
+      try {
+        await fn();
+        setError("");
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log(e);
+        setError(e instanceof Error ? e.message : "unknown error");
+      }
+    };
+
+  const getKeplrAccount = withErrorReporting(async () => {
+    const key = await keplr.getKey(keplrChain);
+    setKeplrAccount(key);
+  });
+
+  const submitIbcTransfer = withErrorReporting(async () => {
+    const wrapperTxProps = {
+      token: namadaChain.data!.nativeTokenAddress,
+      feeAmount: BigNumber(0),
+      gasLimit: BigNumber(20_000),
+      chainId: namadaChain.data!.chainId,
+      publicKey: namadaAccount.data!.publicKey,
+      memo: `Namada->${keplrChain}`,
+    };
+
+    const sdk = await getSdkInstance();
+    const tx = await sdk.tx.buildIbcTransfer(wrapperTxProps, {
+      source: namadaAccount.data!.address,
+      receiver: keplrAccount!.bech32Address,
+      token: namadaChain.data!.nativeTokenAddress,
+      amount: amount!,
+      portId: "transfer",
+      channelId: channel,
+      timeoutHeight: undefined,
+      timeoutSecOffset: undefined,
+      memo: `Namada->${keplrChain}`, // how is this different from the wrapper memo?
```

Comment:
The wrapper memo in this case will be applied with the Tx on Namada, but the IBC Transfer memo gets passed on via the IBC Msg sent to destination, AFAIK. I think in this case, we could omit the wrapper memo if you like. It is kind of weird that we have memos at different levels, the Tx wrapper args, then individual commitments all have their own memo, but in this case, the IbcTransfer memo would be useful to see on Keplr!

Reply:
Ah, I see! I've just gone ahead and removed the memos for now, since they're not used yet, but I can add them in again later.